src/lab/Task8.java

```diff
@@ -0,0 +1,364 @@
+                    4. Retrieve statistic about the movies
+                    5. Find movie by name
+                    6. Update movie rating
+                    7. Delete movie by name
+                    8. Sort the movies
+                    """);
+            userInput = sc.nextInt();
+            switch (userInput) {
+                case 1: {
+                    movies = copyArray(movies, movieArraySize + 3);
+                    System.out.println("""
+                            You selected to insert 3 new movies
+                            
+                            """);
+                    insertMovie(sc, movies, movieArraySize);
+                    movieArraySize += 3;
+                    break;
+                }
+                case 2: {
+                    System.out.println("""
+                            
+                            You selected to display the movies!""");
+                    if (movies.length == 0) {
+                        System.out.println("Unfortunately, there is no movie to display :(");
+                    } else {
+                        System.out.println("""
+                                
+                                Here is the list of movies:
+                                """);
+                        for (Movie movie : movies) {
+                            System.out.println(movie.toString());
+                        }
+                    }
+                    break;
+                }
+                case 3: {
+                    System.out.println("""
+                            
+                            You selected to insert new movies!""");
+                    System.out.println("""
+                            How many new movies you would like to insert?
+                            """);
+                    int numberOfNewNumbers = sc.nextInt();
+                    movies = copyArray(movies, movieArraySize + numberOfNewNumbers);
+                    insertMovie(sc, movies, movieArraySize);
+                    movieArraySize += numberOfNewNumbers;
+                    break;
+                }
+                case 4: {
+                    System.out.println("""
+                            
+                            You selected to view the statistics!""");
+
+
+                    // Finding MAX using Stream
+                    /**
+                     Movie movieWithMaxDouble = Arrays.stream(movies)
+                     .max(Comparator.comparingDouble(Movie::getRating))
+                     .orElse(null);
+                     System.out.println(movieWithMaxDouble.toString());
+                     **/
+
+                    if (movies.length == 0) {
+                        System.out.println("""
+                                
+                                Unfortunately there is no movie to retrieve statistics :(""");
+                        break;
+                    } else if (movies.length == 1) {
```

Comment:
if you do changes, which I mentioned for finding these statistics, then no need this check, method can be callet to find statistics, and response of this method will be same with movies[0]